### PR TITLE
CKEditor Blog Post

### PR DESCRIPTION
**Pull request by tannavee**

### CODE REVIEW

File: content/jupyter-summer-2020.rst [hunk 1]
```diff
@@ -0,0 +1,151 @@
+Developing a Textbook Editor Plugin Inserting Executable Code Blocks
```
Comment (moorepants) [inline content/jupyter-summer-2020.rst:1]:
```suggestion
Developing a LibreTexts Editor Plugin For Inserting Executable Code Blocks
```

File: content/jupyter-summer-2020.rst [hunk 2]
```diff
@@ -0,0 +1,151 @@
+Developing a Textbook Editor Plugin Inserting Executable Code Blocks
+====================================================================
+
+:date: 2020-06-13 00:00:00 
+:tags: oer, education, jupyter, textbooks, engineering, libretexts 
+:category: education 
+:slug: jupyter-summer-2020
```
Comment (moorepants) [inline content/jupyter-summer-2020.rst:7]:
```suggestion
:slug: libretexts-jupyter-plugin

```

File: content/jupyter-summer-2020.rst [hunk 3]
```diff
@@ -0,0 +1,151 @@
+Developing a Textbook Editor Plugin Inserting Executable Code Blocks
+====================================================================
+
+:date: 2020-06-13 00:00:00 
+:tags: oer, education, jupyter, textbooks, engineering, libretexts 
+:category: education 
+:slug: jupyter-summer-2020
+:authors: Hao Huang, Tannavee Kumar, Celine Liang 
+:summary: Blog post on creating a CKEditor plugin which could insert 
+  executable code blocks
```
Comment (moorepants) [inline content/jupyter-summer-2020.rst:10]: Needs to be indented to the B in Blog.

File: content/jupyter-summer-2020.rst [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+:summary: Blog post on creating a CKEditor plugin which could insert 
+  executable code blocks
+
+Background
+----------
+
+During the end of the Fall Quarter 2019 and Winter Quarter 2020, we focused
+on building a `CKEditor Binder Plugin
+<https://github.com/LibreTexts/ckeditor-binder-plugin>`__ to be used on
+the `LibreTexts <https://libretexts.org>`__ to allow textbook authors to
```
Comment (moorepants) [inline content/jupyter-summer-2020.rst:18]:
```suggestion
the `LibreTexts <https://libretexts.org>`__ website to allow textbook authors to

```

File: content/jupyter-summer-2020.rst [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+insert executable code blocks. We intend for both textbook authors and
+readers to have the ability to edit and run code blocks efficiently, making
+online educational content more interactive.
+ 
+Build Process
+-------------
+We started the project by surveying
+LibreTexts authors and readers on their most requested features.
```
Comment (moorepants) [inline content/jupyter-summer-2020.rst:26]: I think we shared this publicly, you could link to it here.

File: content/jupyter-summer-2020.rst [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+
+BinderHub
+^^^^^^^^^
+On the backend, the CKEditor Binder plugin utilizes a project called
+`BinderHub  <https://binderhub.readthedocs.io/en/latest/>`__ to run code
+blocks. BinderHub is developed as part of the Jupyter project and gives
+custom computing environments based on a list of requirements specified
+through a GitHub repository. We intend to use a BinderHub that we deployed
+on our bare-metal Kubernetes cluster as our backend environment to run
+codeblocks.
```
Comment (moorepants) [inline content/jupyter-summer-2020.rst:36]: Remove last sentence.

File: content/jupyter-summer-2020.rst [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+
+Thebelab
+^^^^^^^^
+`Thebelab <https://github.com/minrk/thebelab>`__ and  `Juniper
+<https://github.com/ines/juniper>`__ are two examples of projects which can
+insert code blocks into HTML pages and running them by requesting a kernel
```
Comment (moorepants) [inline content/jupyter-summer-2020.rst:42]:
```suggestion
insert Binder backed code blocks into HTML pages and running them by requesting a kernel

```

File: content/jupyter-summer-2020.rst [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+from a computing backend like BinderHub. We deliberated on which project to
+incorporate into our plugin, and decided to use Thebelab since it seemed to
+be more actively maintained and had more community support.
```
Comment (moorepants) [inline content/jupyter-summer-2020.rst:45]: I though there were some features we preferred.

File: content/jupyter-summer-2020.rst [hunk 9]
```diff
@@ -0,0 +1,151 @@
...
+
+Creating the plugin
+^^^^^^^^^^^^^^^^^^^
+Our plugin is based on the `CKEditor 4
+<https://ckeditor.com/docs/ckeditor4/latest/>`__, an open source “what you
+see is what you get” text editor. This editor is used on the LibreTexts
+website through a service called Mindtouch.
```
Comment (moorepants) [inline content/jupyter-summer-2020.rst:52]: I'd just say, "this is the editor authors use on the LibreTexts website"

File: content/jupyter-summer-2020.rst [hunk 10]
```diff
@@ -0,0 +1,151 @@
...
+
+Our approach to this plugin is to make use of a `widget
+<https://ckeditor.com/docs/ckeditor4/latest/guide/widget_sdk_intro.html>`__,
+on the editor which allows us to place all the HTML elements of Thebelab
+together as one unit. Additionally, we created a `dialog window
```
Comment (moorepants) [inline content/jupyter-summer-2020.rst:57]: Define widget in this context.

File: content/jupyter-summer-2020.rst [hunk 11]
```diff
@@ -0,0 +1,151 @@
...
+<https://ckeditor.com/docs/ckeditor4/latest/guide/dev_howtos_dialog_windows.html>`__
+for each code block so that users can modify each block whenever they want.
+
+Mindtouch Specific Settings
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+One of the challenges we faced was working around Mindtouch, which
```
Comment (moorepants) [inline content/jupyter-summer-2020.rst:63]: Define what Mindtouch is and why it is important to discuss.

File: content/jupyter-summer-2020.rst [hunk 12]
```diff
@@ -0,0 +1,151 @@
...
+Require.js to Mindtouch would break all the JQuery plugins. In order to fix
+all the javascript conflicts, we created registerPlugin.js to include all
+Mindtouch specific js code.
+
+
+Including different programming languages
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+During development, we used `environments developed by the Jupyter project
+<https://github.com/binder-examples>`__. Using their Binder environments
+helped us test and include different languages in our editor. Ideally, we
```
Comment (moorepants) [inline content/jupyter-summer-2020.rst:83]: Eventually we will utilize ...

File: content/jupyter-summer-2020.rst [hunk 13]
```diff
@@ -0,0 +1,151 @@
...
+Packages and libraries can be exported as they normally would; however, if
+an author finds that a specific package or library that they would like to
+use is not currently available they can make a request to have it added by
+either sending an `email to the Jupyter Team <jupyterteam@ucdavis.edu>`__,
+linked in the dialog box, or `open an issue 
+<https://github.com/LibreTexts/ckeditor-binder-plugin/issues>`__ . 
+
+Once the author is ready to insert the code block into their textbook page,
+they have the option to either ‘Insert with code and output,’ ‘Insert with
+code only,’ or ‘Insert with output only.’ Selection of any of these choices
```
Comment (moorepants) [inline content/jupyter-summer-2020.rst:128]: Single ticks in rst do not do render in any special way. Did you want double backticks for literal?

File: content/jupyter-summer-2020.rst [hunk 14]
```diff
@@ -0,0 +1,151 @@
...
+   :align: center
+
+   *Figure 1: Example of what and author can do with CKEditor, and what the output will look like*
+
+
+Future
+------ 
+
+As mentioned before, a long term goal is to use ``default-env`` for the
+packages, this is the same environment that is used for JupyterHub, and by
```
Comment (moorepants) [inline content/jupyter-summer-2020.rst:146]:
```suggestion
packages, this is the same environment that is used for the LIbreTexts-UCD JupyterHub, and by

```

File: content/jupyter-summer-2020.rst [hunk 15]
```diff
@@ -0,0 +1,151 @@
+Developing a Textbook Editor Plugin Inserting Executable Code Blocks
+====================================================================
+
+:date: 2020-06-13 00:00:00
```
Comment (moorepants) [inline content/jupyter-summer-2020.rst:4]: Change  date to date of publishing when ready to merge.

**Comment by moorepants** [comment]
This looks great. I've added comments. Address those and we can merge.

File: content/jupyter-summer-2020.rst (hunk 13 above)
Comment (tannavee) [reply content/jupyter-summer-2020.rst:130]: The single quotes are intentional, as we were quoting what shows up on the dialog box.

Comment (moorepants) [reply content/jupyter-summer-2020.rst:130]: Those are backticks though. If you want single quotes use `'`.

File: content/jupyter-summer-2020.rst (hunk 5 above)
Comment (tannavee) [reply content/jupyter-summer-2020.rst:28]: added this link: https://docs.google.com/document/d/1eV08l_4djKJ7bc8r0LPbD5bp3QT7mHTZgABUleH15H0/edit

File: content/jupyter-summer-2020.rst [hunk 16]
```diff
@@ -0,0 +1,160 @@
+Developing a LibreTexts Editor Plugin For Inserting Executable Code Blocks
+====================================================================
```
Comment (moorepants) [inline content/jupyter-summer-2020.rst:2]:
```suggestion
==========================================================================

```

File: content/jupyter-summer-2020.rst [hunk 17]
```diff
@@ -0,0 +1,161 @@
+Developing a LibreTexts Editor Plugin For Inserting Executable Code Blocks
+==========================================================================
+
+
+:date: 2020-06-22 00:00:00 
+:tags: oer, education, jupyter, textbooks, engineering, libretexts 
+:category: education 
+:slug: libretexts-jupyter-plugin
+
```
Comment (moorepants) [inline content/jupyter-summer-2020.rst:9]: I'm not sure this empty line is allowed.

**Comment by moorepants** [comment]
Looks good! Thanks!

**Pull request merged**